cmd/root.go

```diff
@@ -50,6 +51,25 @@ func init() {
 		}
 	} else {
 		fmt.Fprintf(os.Stderr, "Using configuration file: %s\n", viper.ConfigFileUsed())
+
+	}
+	validateConfiguration()
+}
+
+func validateConfiguration() {
+	u, _ := url.Parse(viper.GetString(argRootServer))
+	if u.Scheme == "" {
+		viper.Set(argRootServer, "https://"+viper.GetString(argRootServer))
+		u, _ = url.Parse(viper.GetString(argRootServer))
+		log.Info("Protocol is not specified, HTTPS is used by default.")
+	}
+	if u.Scheme == "http" {
+		viper.Set(argRootServer, "https://"+u.Host)
+		log.Info("HTTPS is forcely used instead of HTTP.")
+	}
+	if u.Host == "mender.io" {
+		viper.Set(argRootServer, "https://hosted.mender.io")
+		log.Info("Server FQDN is replaced to hosted.mender.io")
 	}
```

Comment:
Honestly, I don't think this is a good idea.
Uploading an artifact using mender.io as server should just fail in a meaningful way.

Reply:
Now it wails with `1` exit code and message starts from:
```
VERBOSE response:
HTTP/1.1 413 Request Entity Too Large
...
```
Isn't it meaningful enough? If not, then what would you suggest?

Reply:
```
66.00 MiB / 256.03 MiB [----------------------------------------->______________________________________________________________________________________________________________________] 25.78% 33.89 MiB p/sVERBOSE response:
HTTP/1.1 413 Request Entity Too Large
Connection: close
Content-Length: 176
Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
Content-Type: text/html
Date: Tue, 19 Jul 2022 10:55:51 GMT
Server: nginx
Via: 1.1 google

<html>
<head><title>413 Request Entity Too Large</title></head>
<body>
<center><h1>413 Request Entity Too Large</h1></center>
<hr><center>nginx</center>
</body>
</html>

FAILURE: artifact upload to 'mender.io' failed with status 413
```

IMO looks better than:
```
63.84 MiB / 256.03 MiB [--------------------------------------->________________________________________________________________________________________________________________________] 24.94% 30.52 MiB p/sVERBOSE response:
HTTP/1.1 413 Request Entity Too Large
Connection: close
Content-Length: 176
Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
Content-Type: text/html
Date: Tue, 19 Jul 2022 10:42:01 GMT
Server: nginx
Via: 1.1 google

<html>
<head><title>413 Request Entity Too Large</title></head>
<body>
<center><h1>413 Request Entity Too Large</h1></center>
<hr><center>nginx</center>
</body>
</html>

FAILURE: artifact upload failed with status 413, reason: <html>
<head><title>413 Request Entity Too Large</title></head>
<body>
<center><h1>413 Request Entity Too Large</h1></center>
<hr><center>nginx</center>
</body>
</html>
```

Reply:
Hope this is what is expected because otherwise I'm out of ideas...

Reply:
@0lmi can you try with a small artifact? The bug report was about receiving an "OK" uploading to mender.io; I suspect the failure you receive now is because you are uploading a large artifact (bigger than 2MB).

Reply:
```
/go/src/github.com/mendersoftware/mender-cli # ./mender-cli artifacts upload dummy_artifact.mender
Using configuration file: /go/src/github.com/mendersoftware/mender-cli/.mender-clirc
0 B / 6.00 KiB [_________________________________________________________________________________________________________________________________________________________________________________] 0.00% ? p/sProcessing uploaded file. This may take around one minute.

6.00 KiB / 6.00 KiB [----------------------------------------------------------------------------------------------------------------------------------------------------------------->] 100.00% 11.97 KiB p/sFAILURE: artifact upload to 'mender.io' failed with status 404
/go/src/github.com/mendersoftware/mender-cli #
/go/src/github.com/mendersoftware/mender-cli # echo $?
1
/go/src/github.com/mendersoftware/mender-cli #
```

with the old error message style it was a lot of html in the output and most likely it was confusing